### ISSUE TITLE
Better error message when spark_home is missing in spark_connect

### ISSUE THREAD

**Issue by javierluraschi** [rstudio/sparklyr#627]
`spark_connect(master = "spark://localhost:7077")` yields and exception rather than a `spark_home` parameter missing error message.

**Comment by javierluraschi** [comment]
I think this is simple enough that is worth trying to fix before `0.6`.

**Comment by javierluraschi** [comment]
Can't repro this one...

```r
spark_connect(master = "spark://localhost:7077")
Error in shell_connection(master = master, spark_home = spark_home, app_name = app_name,  : 
  Failed to connect to Spark (SPARK_HOME is not set).
```
